plutus-scb/src/Plutus/SCB/Webserver/Types.hs

```diff
@@ -44,7 +44,10 @@ data FullReport t =
     deriving (Show, Eq, Generic)
     deriving anyclass (FromJSON, ToJSON)
 
-newtype ContractSignatureResponse t =
-    ContractSignatureResponse [FunctionSchema FormSchema]
+data ContractSignatureResponse t =
+    ContractSignatureResponse
+        { contractSchemas :: [FunctionSchema FormSchema]
+        , contractState   :: ContractInstanceState t
```

Comment:
I think the schema and the state shouldn't be mixed

* Schema is per-contract, you get the schema for a contract even if that contract doesn't have any running instances
* State is per-instance

Reply:
IMO the cleanest way would be to add a route for the active endpoints of a contract instance (and probably change the existing schema route so that it doesn't require an instance ID)